### PR TITLE
CVS-85000 Implement gather exit node handler

### PR DESCRIPTION
**Pull request by dkalinowski**

### CODE REVIEW

File: src/gatherexitnodeinputhandler.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    it->second.mutable_tensor_content()->resize(size);
+    bufferOut = it->second.mutable_tensor_content()->data();
+    return StatusCode::OK;
+}
+
+Status prepareConsolidatedTensorImpl(::inference::ModelInferResponse* response, char*& bufferOut, const std::string& name, size_t size) {
+    OVMS_PROFILE_FUNCTION();
+    auto* proto = response->add_outputs();
+    proto->set_name(name);
+    auto* content = response->add_raw_output_contents();
```
Comment (atobiszei) [inline src/gatherexitnodeinputhandler.cpp:41]: Here we rely on the fact that DAG configuration was validated correct so we don't check for existence of outputs?

Comment (atobiszei) [reply src/gatherexitnodeinputhandler.cpp:41]: Maybe leave a comment here

File: src/test/gather_node_test.cpp [hunk 2]
```diff
@@ -262,3 +263,95 @@ TEST_F(GatherNodeTest, FullFlowGatherInNonExitNode) {
...
+
+TEST_F(TFSGatherExitNodeInputHandlerTest, BufferHasCorrectSize) {
+    ASSERT_EQ(prepareConsolidatedTensorImpl(&response, buffer, tensorName, requestedBufferSize), StatusCode::OK);
+    auto it = response.mutable_outputs()->find(tensorName);
+    ASSERT_NE(it, response.mutable_outputs()->end());
+    auto& proto = it->second;
+    EXPECT_EQ(proto.mutable_tensor_content()->size(), requestedBufferSize);
+}
+
+TEST_F(TFSGatherExitNodeInputHandlerTest, TensorAlreadyExistsInProto) {
```
Comment (atobiszei) [inline src/test/gather_node_test.cpp:300]: We miss similar test for KFS - we should define the behavior in that case as well

**Review by michalkulakowski — APPROVED** [review]

File: src/gatherexitnodeinputhandler.cpp (hunk 1 above)
Comment (dkalinowski) [reply src/gatherexitnodeinputhandler.cpp:47]: Added check if proto with name already exists for KFS.

File: src/test/gather_node_test.cpp (hunk 2 above)
Comment (dkalinowski) [reply src/test/gather_node_test.cpp:300]: Added unit test for KFS

**Pull request merged**